fs/operations/operations_test.go

```diff
@@ -550,6 +550,22 @@ func TestCat(t *testing.T) {
 	}
 }
 
+// test that Purge fallback does not result in deadlock from concurrently listing and removing
```

Comment:
This test should probably be in the internal tests for memory backend

Reply:
My thought was to check whether any other backends have this same problem (now or in the future) -- although that seems unlikely as this is somewhat specific to `memory`.

Maybe there's a more efficient way to check it? The reason for creating so many files is that this is basically a race condition which would not occur with fewer files.

Reply:
As you've probably noticed I try not to get the integration tests to do too much work! Some backends take several seconds to upload a file so uploading 100 files may increase the run time by 5 minutes for the slow backends.

I think this test is fine, but let's put it in the Internal tests for the memory backend as I think it is unlikely that other backends will have this problem in the future.

Reply:
Ok. Full disclosure: bisync's `volatile` test does something very similar (uploading 100 files) which is how I found this issue originally. Should I put in a condition to skip that test on non-local?

Reply:
Probably a good idea, yes!